### ISSUE TITLE
Using custom edges

### ISSUE THREAD

**Issue by scottb5612** [T-F-S/genealogytree#49]
I can build the example on page 211 of the manual,
but I cannot build my own  graph, which 
results in an error unless I take out the 3rd generation.
 
_./tree.tex:43: ./tree.tex:43: Undefined control sequence.
<argument> \gtr@dat@draw@chi                   
l.43 }
      
? Undefined control sequence.
<argument> \gtr@dat@draw@chi 
l.43 }_
      

Here is my code. I've defined \myedgedraw as in the example

\begin{tikzpicture}
\genealogytree[ template=signpost,
% These work
box={halign=center,valign=center},
edges={no background,foreground={blue,Circle-Latex}},
%
% These do not
%box={enhanced jigsaw,opacityback=0.75},
%edges={ custom={\myedgedraw}{\myedgedraw}{\myedgedraw}{\myedgedraw}, },
]
%%%
{
 child{
    g[male]{GG1} p[female]{GG2}
    child[]{ g[female]{G1} p[male]{G2} }
  }
}

**Comment by T-F-S** [comment]
In your example, there a no childs for the 3rd generation. Therefore, the children etoolbox macro is not defined causing the error.

To avoid the problem, the `\myedgedraw` code should be made more fail-safe:

```latex
\documentclass{article}
\usepackage{lmodern,incgraph}
\usepackage[all]{genealogytree}

\newcommand{\myedgedraw}[4]{%
  % parents (#1):
  \ifdef{#1}{%
    \renewcommand*{\do}[1]{
      \draw[#4] (##1)--(#3);
      \path[draw=green!50!black,fill=green!30] (##1) circle (3pt);}%
    \dolistloop{#1}%
  }{}%
  % children (#2):
  \ifdef{#2}{%
    \renewcommand*{\do}[1]{
      \draw[#4] (##1)--(#3);
      \path[draw=yellow!50!black,fill=yellow!50] (##1) circle (3pt);}%
    \dolistloop{#2}%
  }{}%
  % family core (#3):
  \path[draw=purple!50!black,fill=purple!50] (#3) circle (3pt);
}

\begin{document}

\begin{tikzpicture}
\genealogytree[ template=signpost,
  box={enhanced jigsaw,opacityback=0.75},
  edges={
    custom={\myedgedraw}{\myedgedraw}{\myedgedraw}{\myedgedraw},
  },
]
%%%
{
  child
  {
    g[male]{GG1}
    p[female]{GG2}
    child[]
    {
      g[female]{G1}
      p[male]{G2}
    }
  }
}
\end{tikzpicture}

\end{document}
```
![grafik](https://github.com/T-F-S/genealogytree/assets/7982989/a2624737-211d-4ee0-8ec0-ebf2cc24b099)

I will update the documentation example with this enhanced code for the next version.

**Comment by scottb5612** [comment]
Many thanks for working this out ...

I have a question. Is there a way to suppress parts of the tree in order to
display, say a few generations,
starting from an internal node? It would come in very handy.
 I could write parser to do this, that is, I could read the graph into a
tree and then reverse compile the back to the original graph, but a cleaner
solution would be better.

**Comment by T-F-S** [comment]
> I have a question. Is there a way to suppress parts of the tree in order to display, say a few generations, starting from an internal node? It would come in very handy.

Yes, there is. You find everything about it in the documention from page 117 in Section **5.11 Ignoring Input**, e.g. `/gtr/ignore level` to ignore generations.

**Comment by scottb5612** [comment]
Is there a way to select an internal node from which to start generating a
tree?

**Comment by T-F-S** [comment]
> Is there a way to select an internal node from which to start generating a tree?

No, the root node (proband) is fixed.

**Comment by T-F-S** [comment]
Done with https://github.com/T-F-S/genealogytree/releases/tag/v2.4.0
